recipes/occt/all/conanfile.py

```diff
@@ -0,0 +1,202 @@
+
+        tk_libs = self.deps_cpp_info["tk"].libs
+        tk_lib = next(filter(lambda lib: "tk8" in lib, tk_libs))
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "adm/cmake/tk.cmake"),
+            "${CSF_TclTkLibs}",
+            tk_lib)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["3RDPARTY_TCL_LIBRARY_DIR"] = \
+            os.path.join(self.deps_cpp_info["tcl"].rootpath, "lib")
+        self._cmake.definitions["3RDPARTY_TCL_INCLUDE_DIR"] = \
+            self.deps_cpp_info["tcl"].include_paths[0]
+        self._cmake.definitions["3RDPARTY_TK_LIBRARY_DIR"] = \
+            os.path.join(self.deps_cpp_info["tk"].rootpath, "lib")
+        self._cmake.definitions["3RDPARTY_TK_INCLUDE_DIR"] = \
+            self.deps_cpp_info["tk"].include_paths[0]
+        if not self.options.shared:
+            self._cmake.definitions["BUILD_LIBRARY_TYPE"] = "Static"
+
+        self._cmake.definitions["INSTALL_DIR_BIN"] = "bin"
+        self._cmake.definitions["INSTALL_DIR_INCLUDE"] = "include"
+        self._cmake.definitions["INSTALL_DIR_LIB"] = "lib"
+        self._cmake.definitions["INSTALL_DIR_RESOURCE"] = "res/resource"
+        self._cmake.definitions["INSTALL_DIR_DATA"] = "res/data"
+        self._cmake.definitions["INSTALL_DIR_SAMPLES"] = "res/samples"
+        self._cmake.definitions["INSTALL_DIR_DOC"] = "res/doc"
+        self._cmake.definitions["INSTALL_DIR_LAYOUT"] = "Unix"
+
+        self._cmake.configure(source_folder=self._source_subfolder)
+        return self._cmake
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _replace_package_folder(self, source, target):
+        new_name = ""
+        if os.path.isdir(os.path.join(self.package_folder, source)):
+            tools.rmdir(os.path.join(self.package_folder, target))
+            tools.rename(
+                os.path.join(self.package_folder, source),
+                os.path.join(self.package_folder, target))
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy(
+            "LICENSE_LGPL_21.txt",
+            src=self._source_subfolder,
+            dst="licenses")
+        self.copy(
+            "OCCT_LGPL_EXCEPTION.txt",
+            src=self._source_subfolder,
+            dst="licenses")
+        tools.rmdir(os.path.join(self.package_folder, "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib/cmake"))
+        if self.settings.build_type == "Debug":
+            self._replace_package_folder("libd", "lib")
+            self._replace_package_folder("bind", "bin")
+        elif self.settings.build_type == "RelWithDebInfo":
+            self._replace_package_folder("libi", "lib")
+            self._replace_package_folder("bini", "bin")
+
+    def package_info(self):
```

Comment:
```suggestion
    def package_info(self):
        self.cpp_info.names["cmake_find_package"] = "OpenCASCADE"
        self.cpp_info.names["cmake_find_package_multi"] = "OpenCASCADE"
```

OpenCascade provides an official CMake config file with several imported targets, components should be added.

```
-- Installing: /Users/jenkins/w/cci_PR-4094@2/.conan/data/occt/7.5.0/_/_/package/2fcb26941b676939107cc403b75935c9eb3e9858/lib/cmake/opencascade/OpenCASCADEConfig.cmake
-- Installing: /Users/jenkins/w/cci_PR-4094@2/.conan/data/occt/7.5.0/_/_/package/2fcb26941b676939107cc403b75935c9eb3e9858/lib/cmake/opencascade/OpenCASCADEConfigVersion.cmake
-- Installing: /Users/jenkins/w/cci_PR-4094@2/.conan/data/occt/7.5.0/_/_/package/2fcb26941b676939107cc403b75935c9eb3e9858/lib/cmake/opencascade/OpenCASCADEFoundationClassesTargets.cmake
-- Installing: /Users/jenkins/w/cci_PR-4094@2/.conan/data/occt/7.5.0/_/_/package/2fcb26941b676939107cc403b75935c9eb3e9858/lib/cmake/opencascade/OpenCASCADEFoundationClassesTargets-debug.cmake
-- Installing: /Users/jenkins/w/cci_PR-4094@2/.conan/data/occt/7.5.0/_/_/package/2fcb26941b676939107cc403b75935c9eb3e9858/lib/cmake/opencascade/OpenCASCADEModelingDataTargets.cmake
-- Installing: /Users/jenkins/w/cci_PR-4094@2/.conan/data/occt/7.5.0/_/_/package/2fcb26941b676939107cc403b75935c9eb3e9858/lib/cmake/opencascade/OpenCASCADEModelingDataTargets-debug.cmake
-- Installing: /Users/jenkins/w/cci_PR-4094@2/.conan/data/occt/7.5.0/_/_/package/2fcb26941b676939107cc403b75935c9eb3e9858/lib/cmake/opencascade/OpenCASCADEModelingAlgorithmsTargets.cmake
-- Installing: /Users/jenkins/w/cci_PR-4094@2/.conan/data/occt/7.5.0/_/_/package/2fcb26941b676939107cc403b75935c9eb3e9858/lib/cmake/opencascade/OpenCASCADEModelingAlgorithmsTargets-debug.cmake
-- Installing: /Users/jenkins/w/cci_PR-4094@2/.conan/data/occt/7.5.0/_/_/package/2fcb26941b676939107cc403b75935c9eb3e9858/lib/cmake/opencascade/OpenCASCADEVisualizationTargets.cmake
-- Installing: /Users/jenkins/w/cci_PR-4094@2/.conan/data/occt/7.5.0/_/_/package/2fcb26941b676939107cc403b75935c9eb3e9858/lib/cmake/opencascade/OpenCASCADEVisualizationTargets-debug.cmake
-- Installing: /Users/jenkins/w/cci_PR-4094@2/.conan/data/occt/7.5.0/_/_/package/2fcb26941b676939107cc403b75935c9eb3e9858/lib/cmake/opencascade/OpenCASCADEApplicationFrameworkTargets.cmake
-- Installing: /Users/jenkins/w/cci_PR-4094@2/.conan/data/occt/7.5.0/_/_/package/2fcb26941b676939107cc403b75935c9eb3e9858/lib/cmake/opencascade/OpenCASCADEApplicationFrameworkTargets-debug.cmake
-- Installing: /Users/jenkins/w/cci_PR-4094@2/.conan/data/occt/7.5.0/_/_/package/2fcb26941b676939107cc403b75935c9eb3e9858/lib/cmake/opencascade/OpenCASCADEDataExchangeTargets.cmake
-- Installing: /Users/jenkins/w/cci_PR-4094@2/.conan/data/occt/7.5.0/_/_/package/2fcb26941b676939107cc403b75935c9eb3e9858/lib/cmake/opencascade/OpenCASCADEDataExchangeTargets-debug.cmake
-- Installing: /Users/jenkins/w/cci_PR-4094@2/.conan/data/occt/7.5.0/_/_/package/2fcb26941b676939107cc403b75935c9eb3e9858/lib/cmake/opencascade/OpenCASCADEDrawTargets.cmake
-- Installing: /Users/jenkins/w/cci_PR-4094@2/.conan/data/occt/7.5.0/_/_/package/2fcb26941b676939107cc403b75935c9eb3e9858/lib/cmake/opencascade/OpenCASCADEDrawTargets-debug.cmake
```

Reply:
Since this one is fairly complicated can we start with a FIXME??

Reply:
I would prefer to see a proper package_info in the initial PR, otherwise it complelty breaks normal usage of this lib in CMake projects. And also I'm pretty sure that I'll have to fix that myself again.

Reply:
Sadly I share your disappointment.

The components work (API and caveats) is complicated enough it may discourage some contributors (_global statement out of context of this pr_)... some cases are easy it just two or four lines but this, OpenCV, or Qt are very involved.

I'd rather see it merged working, with some conan specific usage, that left unattended to go stale.
That's my motivation for moving it forward without it.

If someone needs it in the future, who's familiar with the project, they can add the components